scripts/test/Muon/elemental_analysis/elemental_analysis_test.py

```diff
@@ -0,0 +1,500 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+from __future__ import print_function, absolute_import
+
+import unittest
+
+from mantid.py3compat import mock
+from mantidqt.utils.qt.testing import start_qapplication
+
+from qtpy.QtGui import QCloseEvent
+
+import matplotlib
+from Muon.GUI.ElementalAnalysis.elemental_analysis import ElementalAnalysisGui
+from Muon.GUI.ElementalAnalysis.elemental_analysis import gen_name
+from MultiPlotting.multi_plotting_widget import MultiPlotWindow
+from MultiPlotting.multi_plotting_widget import MultiPlotWidget
+from MultiPlotting.label import Label
+
+
+@start_qapplication
+class ElementalAnalysisTest(unittest.TestCase):
+    @classmethod
+    def setUpClass(self):
+        super(ElementalAnalysisTest, self).setUpClass()
+        self.gui = ElementalAnalysisGui()
+
+    @classmethod
+    def tearDownClass(self):
+        self.gui = None
+
+    def setUp(self):
+        self.gui.plot_window = None
+        self.gui.color_index = 0
+        self.gui.element_lines = {}
+        self.has_raise_ValueError_once_been_called = False
+
+    def raise_ValueError_once(self):
+        if not self.has_raise_ValueError_once_been_called:
+            self.has_raise_ValueError_once_been_called = True
+            raise ValueError()
+
+    def test_that_default_colour_cycle_is_used(self):
+        cycle = len(matplotlib.rcParams['axes.prop_cycle'])
+        self.assertEqual(cycle, self.gui.num_colors)
+
+        expected = ['C%d' % i for i in range(cycle)]
+        returned = [self.gui.get_color() for _ in range(cycle)]
+        self.assertEqual(expected, returned)
+
+    def test_color_is_increased_every_time_get_color_is_called(self):
+        self.assertEqual(self.gui.color_index, 0)
+        self.gui.get_color()
+        self.assertEqual(self.gui.color_index, 1)
+
+    def test_that_color_index_wraps_around_when_end_reached(self):
+        self.gui.color_index = self.gui.num_colors - 1
+        self.gui.get_color()
+
+        self.assertEqual(self.gui.color_index, 0)
+
+    def test_that_closing_with_no_plot_will_not_throw(self):
+        self.gui.plot_window = None
+        self.gui.closeEvent(QCloseEvent())
```

Comment:
can use the assertRaisesNothing test helper
https://github.com/mantidproject/mantid/blob/master/Framework/PythonInterface/test/python/plugins/algorithms/ReflectometryReductionOneLiveDataTest.py#L15